lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -110,7 +118,12 @@ def create
             args.push('--type', @resource[:type])
         end
 
-        args << @resource[:volume_group]
+        if @resource[:thin]
+            args.push('--thin')
+            args << @resource[:volume_group] + "/" + @resource[:name]
```

Comment:
according `lvcreate --help` the syntax is `lvceate ... --thin VolumeGroupName/PoolLogicalVolume` the `:name` here is IMO the name of the lv to be created not the pool lv.

the `-n (--name)` is IMO still required and should be set to `:name`